app/questionnaire/questionnaire_schema.py

```diff
@@ -1241,6 +1265,14 @@ def _get_section_and_block_ids_dependencies_for_progress_source_and_answer_ids_f
                 calculated_summary_block  # type: ignore
             )
             dependent_answer_ids.update(calculated_summary_answer_ids)
+        elif source == "grand_calculated_summary" and identifier:
```

Comment:
This begs for `identifier` being checked beforehand. Since you check for it in every condition?

Reply:
Good idea - I'm adding this to the other PR (Simplify when rules) because in that one the rules that get checked all have identifier, so the check was never necessary at all, and if I do a check here it will just get deleted again the moment the other card merges